linux/network_device_ops/enable_new_ethernet.yml

```diff
@@ -57,53 +57,88 @@
     - debug: var=network_config.stdout_lines
   when: guest_os_ansible_distribution not in ["VMware Photon OS", "Astra Linux (Orel)"]
 
-# Apply netplan network config changes in Ubuntu server
+# Check new ethernet device status to see whether it has been bring up automatically
+- include_tasks: ../utils/ethernet_ip_get.yml
+  vars:
+    all_ip: False
+
+- name: "Set fact of ethernet device status and IP"
+  set_fact:
+    eth_dev_status: "{{ ethernets[eth_dev] if ethernets | length > 0 else {} }}"
+
 - block:
-    - name: "Apply netplan configuration file for new added nic {{ eth_dev }}"
-      command: "netplan apply"
+    - include_tasks: ../utils/get_service_info.yml
+      vars:
+        service_name: "NetworkManager.service"
+
+    # RHEL/SLED/Ubuntu Desktop network interfaces are managed by NetworkManager
+    - name: "Bring link '{{ eth_dev }}' up"
+      shell: "nmcli device connect {{ eth_dev }}"
       delegate_to: "{{ vm_guest_ip }}"
-      become: yes
-    - include_tasks: ../../common/vm_wait_connection.yml
-  when:
-    - guest_os_ansible_distribution == "Ubuntu"
-    - guest_os_with_gui is defined and not guest_os_with_gui
-
-# Bring up the new network interface
-- name: "Bring link '{{ eth_dev }}' up"
-  shell: "nmcli device connect {{ eth_dev }}"
-  delegate_to: "{{ vm_guest_ip }}"
-  register: set_link_down_up
-  when:
-    - guest_os_family == "RedHat"
-    - guest_os_ansible_distribution_major_ver | int >= 8
-
-- name: "Bring link '{{ eth_dev }}' up"
-  shell: "ip link set {{ eth_dev }} down; ip link set {{ eth_dev }} up"
-  delegate_to: "{{ vm_guest_ip }}"
-  register: set_link_down_up
-  when: >
-    (guest_os_ansible_distribution in ["VMware Photon OS", "Flatcar"]) or
-    (guest_os_ansible_distribution == "Ubuntu" and guest_os_with_gui is defined and guest_os_with_gui)
+      when:
+        - service_info
+        - service_info.state is defined
+        - service_info.state in ['active', 'running']
+
+    - block:
+        # Ubuntu server is using netplan to manage network interfaces
+        - block:
+            - name: "Apply netplan configuration file for new added nic {{ eth_dev }} in Ubuntu"
+              command: "netplan apply"
+              delegate_to: "{{ vm_guest_ip }}"
+              become: yes
+            - include_tasks: ../../common/vm_wait_connection.yml
+          when: guest_os_ansible_distribution == "Ubuntu"
+
+        # SLES is using Wicked to manage network interface
+        - block:
+            # Bring down new ethernet device if no IP assigned
+            - name: "Bring link '{{ eth_dev }}' down for {{ guest_os_ansible_distribution }}"
+              shell: "wicked ifdown {{ eth_dev }}"
+              delegate_to: "{{ vm_guest_ip }}"
+              when:
+                - eth_dev_status.status is defined
+                - eth_dev_status.status | lower == 'up'
+                - not eth_dev_status.ipv4
```

Comment:
Is "eth_dev_status.ipv4 is undefined" need to be considered here?

Reply:
Yes. Just updated it and removed the useless unattend config files.